cmd/osmosisd/cmd/forceprune.go

```diff
@@ -128,13 +139,21 @@ func compactBlockStore(dbPath string) error {
 	fmt.Println("Compacting Block Store ...")
 
 	db, err := leveldb.OpenFile(dbPath+"/blockstore.db", &compactOpts)
+	// nolint: staticcheck
 	defer db.Close()
 	if err != nil {
 		return err
 	}
 	if err = db.CompactRange(*util.BytesPrefix([]byte{})); err != nil {
 		return err
 	}
+	// N.B: We duplicate the call to db.Close() on top of
+	// the call in defer statement above to make sure that the resources
+	// are properly released and any potential error from Close()
+	// is handled. Close() should be idempotent so this is acceptable.
+	if err := db.Close(); err != nil {
+		return err
+	}
```

Comment:
I don't get why are we doing this? The defer handles this more cleanly?

Reply:
If we do `defer db.Close()`, it does not do error handling.

The new change validates the error while keeping `defer db.Close()` as well. The code in `defer` runs even if the code panics, making sure that we always close the resources. Calling this twice should not be a problem because `Close()` is idempotent

Reply:
Can we instead do a named return error, (e.g. `func compactBlockStore(dbPath string) (rerr error)`, and then do `defer rerr = db.Close()`. This is the general pattern for handling errors in a panic.

Encouraging db closing being isolated scope / not handled multiple times is ideal imo.